### DISCUSSION TITLE
firefox does not seem to use HW accelerated video?

### DISCUSSION THREAD

**Discussion by tve** [Manjaro-Sway/manjaro-sway#360] [Q&A] [answered]
I have an AMD GPU and firefox doesn't seem to be using the HW accelerated video decoding, e.g. when playing youtube. Running full-screen @4K drops about 1/3 of the frames, pretty shitty experience (and it's not the network). This worked out of the box in the KDE wayland edition for me. I don't remember how to show the HW video decode usage, have to remember what that utility was...
Q: known issue? something due to sway?
I can go and track down stuff for this but thought it would be best to ask first. Have to get some sleep now anyway :-)

```
~> inxi -G
Graphics:
  Device-1: AMD Cezanne [Radeon Vega Series / Radeon Mobile Series]
    driver: amdgpu v: kernel
  Display: wayland server: Xwayland v: 22.1.5 compositor: sway v: 1.7
    driver: gpu: amdgpu resolution: 1: 3840x2160~60Hz 2: 2560x1440~60Hz
  Message: No known Wayland EGL/GBM data sources. API: EGL/GBM
```

**Comment by boredland** [comment]
known, yes. sway issue: I don't know. afaik firefox enables hardware accelerated video only for [specific hardware/driver combinations](https://wiki.archlinux.org/title/Firefox#Hardware_video_acceleration) by default - was it the same device you tried this on?

**Reply by tve** [reply, marked as answer]
It's the same device.

Looking at about:support I see:

> HARDWARE_VIDEO_DECODING	
> available by default
> blocklisted by env: Blocklisted by gfxInfo

I can't find any way to see the "gfxinfo blocklist".

I ended up using the tips in https://discourse.ubuntu.com/t/enabling-accelerated-video-decoding-in-firefox-on-ubuntu-21-04/22081 and now bliss is restored!

**Reply by boredland** [reply]
Would be nice to find out how this happened automatically on KDE tho.

**Reply by boredland** [reply]
I think I followed all of this, but still have no hardware accelerated video on my pretty new intel card.
